osv2/executor.go

```diff
@@ -28,13 +28,16 @@ func NewExecutor(client *opensearch.Client) *Executor {
 //   - The request to OpenSearch fails
 //   - The results json cannot be unmarshalled
 func (e *Executor) MGet(ctx context.Context, req *opensearchtools.MGetRequest) (*opensearchtools.MGetResponse, error) {
-	specMGetRequest := FromModelMGetRequest(req)
+	specMGetRequest, err := FromDomainMGetRequest(req)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
@cs-marshall re
> Do we also want to return a ValidationResults? I would think we want nil if no validation issues.
Not sure what you mean... we are returning a ValidationResults here (or nil if there are validation issues)... can you clarify?

Reply:
@cs-marshall re

> We should add and call Validate on the specMGetRequest as well after the conversion. The Domain model, if valid, does not guarantee that the v2 version will be valid

I would say no since the V2 specific checks (if we had any) would get called here: https://github.com/CrowdStrike/opensearchtools/pull/13/files#diff-eb72ae896f7b8eeef11280b2240c22472117aed40de0be70f540824206fdf29aR101

That seem right?

The counter argument to that is that maybe we do need a `.Validate()` on the osv2.MGetRequest so that if a consumer is using it directly, they will have a way to do the validation.

Reply:
I added a .Validate() to osv2.MGetRequest so that a consumer using osv2 directly has a way to validate but it's not called by any of the executor/domain stuff.